src/Microsoft.OData.Core/DuplicatePropertyNameChecker.cs

```diff
@@ -122,6 +123,25 @@ public void Reset()
         {
             propertyState.Clear();
         }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    Reset();

```

Comment:
Is it necessary to reset the duplicate checker if it's not going to be used again?

Reply:
#Resolved